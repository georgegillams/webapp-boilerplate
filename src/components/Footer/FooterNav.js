import React from 'react';
import PropTypes from 'prop-types';
import { cssModules } from '@george-gillams/components/helpers/cssModules';
import { THEMES } from '@george-gillams/components/Theming';
import TextLink from '@george-gillams/webapp/components/TextLink';

import STYLES from './footer-nav.scss';

const getClassName = cssModules(STYLES);

const TechSpecs = props => {
  const { className, ...rest } = props;

  return (
    <div className={getClassName('footer-nav__container', className)} {...rest}>
      <TextLink className={getClassName('footer-nav__link')} theme={THEMES.allWhite} href="/sitemap">
        Site map
      </TextLink>
      <TextLink className={getClassName('footer-nav__link')} theme={THEMES.allWhite} href="/status">
        Status
      </TextLink>
      <TextLink className={getClassName('footer-nav__link')} theme={THEMES.allWhite} href="/privacy-policy">
        Privacy
      </TextLink>
    </div>
  );
};

TechSpecs.propTypes = {
  className: PropTypes.string,
};

TechSpecs.defaultProps = {
  className: null,
};

export default TechSpecs;
