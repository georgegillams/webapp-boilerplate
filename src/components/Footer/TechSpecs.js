import React from 'react';
import PropTypes from 'prop-types';
import { cssModules } from 'gg-components/helpers/cssModules';
import { Section } from 'gg-components/Typography';

import STYLES from './tech-specs.scss';
import reactLogo from './react-logo.svg';
import reduxLogo from './redux-logo.svg';
import awsLogo from './aws-logo.svg';

const getClassName = cssModules(STYLES); // REGEX_REPLACED

const TechSpecs = props => {
  const { light, fancy, className, ...rest } = props;

  const outerClassNameFinal = [];
  if (className) {
    outerClassNameFinal.push(className);
  }

  const iconClassNameFinal = [getClassName('tech-specs__icon')];
  const darkIconClassNameFinal = [getClassName('tech-specs__icon')];
  iconClassNameFinal.push(getClassName('tech-specs__icon--no-light'));
  darkIconClassNameFinal.push(getClassName('tech-specs__icon--light-inverted'));

  return (
    <div className={outerClassNameFinal.join(' ')} {...rest}>
      <Section className={getClassName('tech-specs__container')} noPadding light={light} fancy={fancy}>
        Built in
        <a href="https://reactjs.org/" rel="noopener noreferrer" target="_blank">
          <img alt="React" width={5} height={5} className={darkIconClassNameFinal.join(' ')} src={reactLogo} />
        </a>
        and
        <a href="https://redux.js.org/" rel="noopener noreferrer" target="_blank">
          <img alt="Redux" width={5} height={5} className={darkIconClassNameFinal.join(' ')} src={reduxLogo} />
        </a>
        Hosted on
        <a href="https://aws.amazon.com/" rel="noopener noreferrer" target="_blank">
          <img
            alt="Amazon Web Services"
            width={8.28}
            height={5}
            className={darkIconClassNameFinal.join(' ')}
            style={{ marginTop: '.8rem', maxWidth: '2.9rem' }}
            src={awsLogo}
          />
        </a>
      </Section>
    </div>
  );
};

TechSpecs.propTypes = {
  light: PropTypes.bool,
  fancy: PropTypes.bool,
  className: PropTypes.string,
};

TechSpecs.defaultProps = {
  light: false,
  fancy: false,
  className: null,
};

export default TechSpecs;
