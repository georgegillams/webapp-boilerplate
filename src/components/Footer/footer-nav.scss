@import '~utils/tokens';

.footer-nav {
  &__container {
    display: flex;
    width: 100%;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
    color: $not-black-invert;
  }

  &__link {
    margin: 0.5rem 0.5rem 0 0.5rem;
    color: $not-black-invert;
  }
}
