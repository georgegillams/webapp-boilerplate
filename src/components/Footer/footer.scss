@import '~utils/tokens';

.footer {
  &__container {
    display: flex;
    height: 15rem;
    padding: 2rem 2rem;
    flex-direction: column;
    justify-content: space-around;
    background-color: $not-black;
    background-position: center;
    background-size: cover;
    color: $not-black-invert;

    @include bpk-breakpoint-mobile {
      padding: 2rem 1rem;
    }

    @media (prefers-color-scheme: dark) {
      border-top: solid white 0.15rem;
      background-color: $background-color-elevated-colored-invert;
    }
  }

  &__logo {
    text-align: center;
  }

  &__tech {
    text-align: center;
  }
}
