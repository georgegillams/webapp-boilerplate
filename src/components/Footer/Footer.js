import React from 'react';
import PropTypes from 'prop-types';
import { cssModules } from 'gg-components/helpers/cssModules';

import Logo from '../Logo';

import TechSpecs from './TechSpecs';
import STYLES from './footer.scss';

const getClassName = cssModules(STYLES); // REGEX_REPLACED

const Footer = props => {
  const { className, ...rest } = props;
  const outerClassNameFinal = [getClassName('footer__container')];
  if (className) {
    outerClassNameFinal.push(className);
  }

  const date = new Date();
  const isPride = date.getMonth() === 5;

  return (
    <footer id="footer" className={outerClassNameFinal.join(' ')} {...rest}>
      <Logo pride={isPride} small alwaysCentered className={getClassName('footer__logo')} padding={false} />
      <TechSpecs className={getClassName('footer__tech')} light />
    </footer>
  );
};

Footer.propTypes = {
  className: PropTypes.string,
};

Footer.defaultProps = {
  className: null,
};

export default Footer;
