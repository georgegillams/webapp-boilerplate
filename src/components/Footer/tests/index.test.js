import React from 'react';

import { render } from '@testing-library/react';

import Footer from '..';

describe('<Footer />', () => {
  it('Should render correctly', () => {
    const { container } = render(<Footer />);

    expect(container).toMatchSnapshot();
  });
});
