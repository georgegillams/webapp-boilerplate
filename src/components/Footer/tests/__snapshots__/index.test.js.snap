// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Should render correctly 1`] = `
<div>
  <footer
    class="footer__container"
    id="footer"
  >
    <div
      class="logo__container footer__logo logo__container--centred logo__container--no-padding"
    >
      <a
        href="/"
      >
        <h1
          aria-label="Home page"
          class="logo__heading logo__heading--pride logo__heading--no-padding"
          role="button"
        >
          LOGO
        </h1>
      </a>
    </div>
    <div
      class="footer__tech"
    >
      <div
        class="typography__main typography--no-padding tech-specs__container"
      >
        Built in
        <a
          href="https://reactjs.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="React"
            class="tech-specs__icon"
            height="5"
            src="IMAGE_MOCK"
            width="5"
          />
        </a>
        and
        <a
          href="https://redux.js.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Redux"
            class="tech-specs__icon"
            height="5"
            src="IMAGE_MOCK"
            width="5"
          />
        </a>
        Hosted on
        <a
          href="https://aws.amazon.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Amazon Web Services"
            class="tech-specs__icon"
            height="5"
            src="IMAGE_MOCK"
            style="margin-top: .8rem; max-width: 2.9rem;"
            width="8.28"
          />
        </a>
      </div>
    </div>
    <div
      class="footer-nav__container footer__nav"
    >
      <a
        class="typography__main typography__link typography__link--text-link typography--no-padding typography--light typography--light--text-link typography--inline footer-nav__link footer__nav"
        href="/sitemap"
      >
        Site map
      </a>
      <a
        class="typography__main typography__link typography__link--text-link typography--no-padding typography--light typography--light--text-link typography--inline footer-nav__link footer__nav"
        href="/status"
      >
        Status
      </a>
    </div>
  </footer>
</div>
`;

exports[`<Footer /> Should render correctly with custom className 1`] = `
<div>
  <footer
    class="footer__container test-custom-className"
    id="footer"
  >
    <div
      class="logo__container footer__logo logo__container--centred logo__container--no-padding"
    >
      <a
        href="/"
      >
        <h1
          aria-label="Home page"
          class="logo__heading logo__heading--pride logo__heading--no-padding"
          role="button"
        >
          LOGO
        </h1>
      </a>
    </div>
    <div
      class="footer__tech"
    >
      <div
        class="typography__main typography--no-padding tech-specs__container"
      >
        Built in
        <a
          href="https://reactjs.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="React"
            class="tech-specs__icon"
            height="5"
            src="IMAGE_MOCK"
            width="5"
          />
        </a>
        and
        <a
          href="https://redux.js.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Redux"
            class="tech-specs__icon"
            height="5"
            src="IMAGE_MOCK"
            width="5"
          />
        </a>
        Hosted on
        <a
          href="https://aws.amazon.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Amazon Web Services"
            class="tech-specs__icon"
            height="5"
            src="IMAGE_MOCK"
            style="margin-top: .8rem; max-width: 2.9rem;"
            width="8.28"
          />
        </a>
      </div>
    </div>
    <div
      class="footer-nav__container footer__nav"
    >
      <a
        class="typography__main typography__link typography__link--text-link typography--no-padding typography--light typography--light--text-link typography--inline footer-nav__link footer__nav"
        href="/sitemap"
      >
        Site map
      </a>
      <a
        class="typography__main typography__link typography__link--text-link typography--no-padding typography--light typography--light--text-link typography--inline footer-nav__link footer__nav"
        href="/status"
      >
        Status
      </a>
    </div>
  </footer>
</div>
`;
